src/com/metricsassignment/metrics/CBOMetric.java

```diff
@@ -1,12 +1,61 @@
 package com.metricsassignment.metrics;
 
+import com.github.javaparser.StaticJavaParser;
 import com.github.javaparser.ast.CompilationUnit;
+import com.metricsassignment.visitors.CouplingBetweenObjectsVisitor;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
 
 // CBO - Coupling between Objects
 public class CBOMetric implements Metric {
 
+    private Map<String, Set<String>> cboMap;
+
+    public CBOMetric(List<File> javaFiles) {
+        this.cboMap = buildStructure(javaFiles);
+    }
+
     @Override
     public double calculate(CompilationUnit compilationUnit) {
-        return 0;
+        String classname = compilationUnit.getType(0).getNameAsString();
+
+        // all references to other classes within this class
+        Integer referenceToOtherClass = cboMap.get(classname).size();
+
+        // every class that references this class
+        Integer referencedInOtherClasses = 0;
+        for( Set<String> refs : cboMap.values()){
+            if(refs.contains(classname)){
+                referencedInOtherClasses++;
+            }
+        }
+
+        return referenceToOtherClass + referencedInOtherClasses;
```

Comment:
This would contain duplicates.

Reply:
True, I put them into a set, so it should only consider them once even if it's a two-way reference.